editor/filesystem_dock.cpp

```diff
@@ -89,34 +90,77 @@ bool FileSystemList::edit_selected() {
 	ERR_FAIL_COND_V_MSG(!is_anything_selected(), false, "No item selected.");
 	int s = get_current();
 	ERR_FAIL_COND_V_MSG(s < 0, false, "No current item selected.");
-	ensure_current_is_visible();
 
-	Rect2 rect;
+	const Rect2 rect = get_item_rect(s, get_icon_mode() == ItemList::ICON_MODE_LEFT);
+	const Ref<StyleBoxFlat> focus_style = get_theme_stylebox(SNAME("focus"), SNAME("ItemList"));
+
+	// Ensure current item is visible.
+	VScrollBar *v_scroll = get_v_scroll_bar();
+	if (v_scroll->is_visible()) {
+		int from = v_scroll->get_value();
+		int to = from + v_scroll->get_page();
+		if (rect.position.y < from) {
+			const int panel_offset = get_theme_stylebox(SNAME("panel"), SNAME("ItemList"))->get_offset().y;
+			const int border_width_top = focus_style->get_border_width(Side::SIDE_TOP);
```

Comment:
This is wrong. `focus_style` does not need to be a StyleBoxFlat (when using custom theme), so this can result in a crash or a wrong margin.
You should use `StyleBox.get_margin()` method instead.